source/python/idetect/scraper.py

```diff
@@ -90,107 +142,81 @@ def format_date(date_string):
     return formatted_date
 
 
-class Scraper(object):
-
-    def __init__(self):
-        pass
+def html_article(url):
+    """Downloads and extracts content plus metadata for html page
+    Parameters
+    ----------
+    url: url of page to be scraped
+
+    Returns
+    -------
+    article: An object of class Article containing the content and metadata.
+    """
+
+    a = newspaper.Article(url)
+    a.download()
+    if a.download_state == 2:
+        a.parse()
+        article_domain = a.source_url
+        article_title = a.title
+        article_authors = a.authors
+        article_pub_date = a.publish_date
+        article_text = remove_newline(a.text)
+        # tag the type of article
+        # currently default to text but should be able to determine img/video
+        # etc
+        article_content_type = 'text'
+        return article_text, article_pub_date, article_title, article_content_type, article_authors, article_domain
+    else:  # Temporary fix to deal with https://github.com/codelucas/newspaper/issues/280
+        return "retrieval_failed", None, "", datetime.datetime.now(), "", ""
+
+
+def get_pdf(url):
+    ''' Takes a pdf url, downloads it and saves it locally.'''
+    try:
+        response = request.urlopen(url)  # not sure if this is needed?
+        publish_date = response.getheader('Last-Modified')
+        pdf_file = open('file_to_convert.pdf', 'wb')
```

Comment:
You should use the `tempfile` module to get a temporary file name. If two workers are simultaneously downloading to the same file, we could get corruption.